broadcaster/assets/js/panel.js

```diff
@@ -111,8 +123,117 @@ async function startStreaming() {
+          timestamp: undefined,
+          bytesSent: 0,
+          packetsSent: 0,
+          retransmittedPacketsSent: 0,
+          nackCount: 0,
+        };
+
+        stats.forEach((report) => {
+          if (report.type === 'outbound-rtp' && report.kind === 'video') {
+            videoReport.timestamp = report.timestamp;
+            videoReport.bytesSent += report.bytesSent;
+            videoReport.packetsSent += report.packetsSent;
+            videoReport.retransmittedPacketsSent +=
+              report.retransmittedPacketsSent;
+            videoReport.nackCount += report.nackCount;
+          } else if (
+            report.type === 'outbound-rtp' &&
+            report.kind === 'audio'
+          ) {
+            audioReport = report;
+          }
+        });
+
+        // calculate bitrates
+        let bitrate;
+        if (!lastVideoReport) {
+          bitrate = (videoReport.bytesSent * 8) / 1000;
+        } else {
+          const timeDiff =
+            (videoReport.timestamp - lastVideoReport.timestamp) / 1000;
+          if (timeDiff == 0) {
+            // this should never happen as we are getting stats every second
+            bitrate = 0;
+          } else {
+            bitrate =
+              ((videoReport.bytesSent - lastVideoReport.bytesSent) * 8) /
+              timeDiff;
+          }
+        }
+
+        videoBitrate.innerText = (bitrate / 1000).toFixed();
+        lastVideoReport = videoReport;
+
+        if (!lastAudioReport) {
+          bitrate = audioReport.bytesSent;
+        } else {
+          const timeDiff =
+            (audioReport.timestamp - lastAudioReport.timestamp) / 1000;
+          if (timeDiff == 0) {
+            // this should never happen as we are getting stats every second
+            bitrate = 0;
+          } else {
+            bitrate =
+              ((audioReport.bytesSent - lastAudioReport.bytesSent) * 8) /
+              timeDiff;
+          }
+        }
+
+        audioBitrate.innerText = (bitrate / 1000).toFixed();
+        lastAudioReport = audioReport;
+
+        // calculate packet loss
+        if (!lastAudioReport || !lastVideoReport) {
+          packetLoss.innerText = 0;
+        } else {
+          const packetsSent =
+            lastVideoReport.packetsSent + lastAudioReport.packetsSent;
+          const rtxPacketsSent =
+            lastVideoReport.retransmittedPacketsSent +
+            lastAudioReport.retransmittedPacketsSent;
+          const nackReceived =
+            lastVideoReport.nackCount + lastAudioReport.nackCount;
+
+          if (nackReceived == 0) {
+            packetLoss.innerText = 0;
+          } else {
+            packetLoss.innerText = (
+              (nackReceived / (packetsSent - rtxPacketsSent)) *
+              100
+            ).toFixed();
+          }
+        }
+      }, 1000);
+    } else if (pc.connectionState === 'disconnected') {
+      console.warn('Peer connection state changed to `disconnected`');
+      setStatusIcon(false);
```

Comment:
`disconnected` doesn't always mean we stopped sending data. It means that the other side stopped responding to our binding requests :wink: